### PR TITLE
Intialize Java server

### PR DESCRIPTION
**Pull request by calvinqc**
Setting up Java  server with: 
- Kafka
- AWS DynamoDB
- MySQL
- Docker
- Springboot

### CODE REVIEW

File: server/src/main/cmpe172/server/controller/StudentController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import server.model.Student;
+import server.repository.StudentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+public class StudentController {
```
Comment (ellamaolson) [inline server/src/main/cmpe172/server/controller/StudentController.java]: Why are we making a student controller?

File: server/src/main/cmpe172/server/model/Student.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package server.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Student {
```
Comment (ellamaolson) [inline server/src/main/cmpe172/server/model/Student.java]: Why do we need a Student class?

File: server/src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,18 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/cmpe172?allowPublicKeyRetrieval=true&useSSL=false&serverTimezone=UTC
+spring.datasource.username=root
+spring.datasource.password=password
+
+app.topic=calvin-nguyen-kafka-spring
```
Comment (ellamaolson) [inline server/src/main/resources/application.properties]: Rename kafka topic to csearch-topic

File: server/src/main/cmpe172/server/service/KafkaConsumer.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class KafkaConsumer {
+
+    public static List<String> messages = new ArrayList<>();
+    private final static String topic = "calvin-nguyen-kafka-spring";
```
Comment (ellamaolson) [inline server/src/main/cmpe172/server/service/KafkaConsumer.java]: Rename topic to csearch-topic

File: server/src/main/cmpe172/server/service/KafkaConsumer.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    private final static String groupId = "kafka-spring-app";
```
Comment (ellamaolson) [inline server/src/main/cmpe172/server/service/KafkaConsumer.java]: rename groupId to csearch-kafka-app

File: server/pom.xml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.3.0.M3</version>
+		<relativePath/> <!-- lookup parent from server.repository -->
+	</parent>
+	<groupId>cmpe172</groupId>
```
Comment (ellamaolson) [inline server/pom.xml]: Rename groupId to something like 172-csearch

**Comment by calvinqc** [comment]
Setup a new server with Docker, @ellamaolson will continue work on setting up RDS with a new setup with Docker.

**Pull request closed without merging**